File: Algorithm-Swift/Algorithm-Swift/Inflearn/23_연속부분증가수열.swift
//
//  23_연속부분증가수열.swift
//  Algorithm-Swift
//
//  Created by SHIN YOON AH on 2023/08/04.
//

import Foundation

func solution23() {
    let n = Int(readLine()!)!
    let nums = readLine()!.split(separator: " ").map { Int($0)! }
    var cnt = 0, max = 0

    for i in stride(from: 0, to: n-1, by: 1) {
        if nums[i+1] - nums[i] >= 0 {
            cnt += 1
            if cnt > max {
                max = cnt
            }
        } else {
            cnt = 0
        }
    }

    if max == 0 {
        print(max)
    } else {
        print(max + 1)
    }
}

File: Algorithm-CPP/23_연속부분증가수열.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n, cnt = 0, max = 0;
    cin >> n;

    vector<int> arr(n);
    for(int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    for(int i = 0; i < n - 1; i++) {
        if(arr[i+1] - arr[i] >= 0) {
            cnt++;
            if(cnt > max) max = cnt;
        } 
        else cnt = 0;
    }

    if(max != 0) cout << max + 1 << endl;
    else cout << max << endl;

    return 0;
}